redash/tasks.py

```diff
@@ -317,3 +318,47 @@ def execute_query(self, query, data_source_id, metadata):
 @celery.task(base=BaseTask)
 def record_event(event):
     models.Event.record(event)
+
+def notify_hipchat(alert, html, new_state):
+    try:
+        hipchat_client = hipchat.HipChat(token=settings.HIPCHAT_API_TOKEN)
+        message = '[' + new_state.upper() + '] ' + alert.name + '<br />' + html
+        hipchat_client.message_room(settings.HIPCHAT_ROOM_ID, settings.NAME, message, message_format='html')
+    except:
+        logger.exception("hipchat send ERROR.")
+
+def notify_mail(alert, html, new_state, app):
+    recipients = [s.email for s in alert.subscribers()]
+    logger.debug("Notifying: %s", recipients)
+    try:
+        with app.app_context():
+            message = Message(recipients=recipients,
+                              subject="[{1}] {0}".format(alert.name, new_state.upper()),
+                              html=html)
+            mail.send(message)
+    except:
+        logger.exception("mail send ERROR.")
+
+def notify_webhook(alert, query, html, new_state):
+    http_client = httplib2.Http(timeout=30)
+    if settings.WEBHOOK_USERNAME:
+        http_client.add_credentials(settings.WEBHOOK_USERNAME, settings.WEBHOOK_PASSWORD)
+    try:
+        data = {
+            'new_state':new_state.upper(),
+            'alert_name':alert.name,
+            'alert_id':alert.id,
+            'query_id':query.id,
+            'url_base':settings.HOST
+        }
```

Comment:
alert.to_dict , I feel that there is very much unnecessary information...

data example

```
{'state': 'triggered', 'options': {u'column': u'count(*)', u'value': 7, u'op': u'greater than'}, 'name': u'fuga: count(*) greater than 7', 'updated_at': datetime.datetime(2015, 11, 18, 1, 35, 37, 973331), 'query': {'latest_query_data_id': 108, 'schedule': None, 'is_archived': False, 'updated_at': datetime.datetime(2015, 11, 17, 4, 10, 40, 182108, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), 'user': {'auth_type': 'password', 'created_at': datetime.datetime(2015, 11, 4, 4, 43, 55, 314074, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), 'name': u'toyama0919', 'gravatar_url': 'https://www.gravatar.com/avatar/1933de2ab395fc2f64a20fae9f77029a?s=40', 'updated_at': datetime.datetime(2015, 11, 17, 1, 58, 3, 611870, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), 'id': 2, 'groups': [u'default'], 'email': u'toyama-h@ipros.jp'}, 'query': u'select count(*) from customer', 'id': 3, 'description': None, 'name': u'fuga', 'created_at': datetime.datetime(2015, 11, 17, 4, 10, 40, 182019, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), 'last_modified_by': {'auth_type': 'password', 'created_at': datetime.datetime(2015, 11, 4, 4, 43, 55, 314074, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), 'name': u'toyama0919', 'gravatar_url': 'https://www.gravatar.com/avatar/1933de2ab395fc2f64a20fae9f77029a?s=40', 'updated_at': datetime.datetime(2015, 11, 17, 1, 58, 3, 611870, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), 'id': 2, 'groups': [u'default'], 'email': u'toyama-h@ipros.jp'}, 'query_hash': u'dd8bb0018aea7d1892151fd9478bfc0e', 'api_key': u'42858af3cdbf80fe90e55d218a74f1f907d1a5d7', 'data_source_id': 2}, 'created_at': datetime.datetime(2015, 11, 18, 1, 35, 31, 868308, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), 'last_triggered_at': None, 'id': 103, 'user': {'auth_type': 'password', 'created_at': datetime.datetime(2015, 11, 4, 4, 43, 55, 314074, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), 'name': u'toyama0919', 'gravatar_url': 'https://www.gravatar.com/avatar/1933de2ab395fc2f64a20fae9f77029a?s=40', 'updated_at': datetime.datetime(2015, 11, 17, 1, 58, 3, 611870, tzinfo=psycopg2.tz.FixedOffsetTimezone(offset=0, name=None)), 'id': 2, 'groups': [u'default'], 'email': u'toyama-h@ipros.jp'}}
```